unit_test/entities/controller/test_controller_class.py

```diff
@@ -0,0 +1,144 @@
+        # Assert
+        self.assertEqual(total_supply, 1.0)
+
+    def test__set_producers_to_max(self):
+        """Test to set the producers to the max power."""
+        # Act
+        producers = self.controller._set_producers_to_max()
+        # Assert
+        self.assertEqual(producers[self.producer1.id][PROPERTY_HEAT_DEMAND], 1.0)
+        self.assertEqual(producers[self.producer1.id][PROPERTY_TEMPERATURE_RETURN], 20.0)
+        self.assertEqual(producers[self.producer1.id][PROPERTY_TEMPERATURE_SUPPLY], 30.0)
+        self.assertTrue(producers[self.producer1.id][PROPERTY_SET_PRESSURE])
+
+        self.assertEqual(producers[self.producer2.id][PROPERTY_HEAT_DEMAND], 2.0)
+        self.assertEqual(producers[self.producer2.id][PROPERTY_TEMPERATURE_RETURN], 40.0)
+        self.assertEqual(producers[self.producer2.id][PROPERTY_TEMPERATURE_SUPPLY], 50.0)
+        self.assertFalse(producers[self.producer2.id][PROPERTY_SET_PRESSURE])
+
+    def test__set_consumer_capped(self):
+        """Test to set the consumer to the capped power."""
+        self.controller.producers[1].power = 0.5
+        # Act
+        consumers = self.controller._set_consumer_capped(datetime.now())
+        # Assert
+        self.assertEqual(consumers[self.consumer1.id][PROPERTY_HEAT_DEMAND], 1.0 * 1.5 / 3.0)
+        self.assertEqual(consumers[self.consumer2.id][PROPERTY_HEAT_DEMAND], 2.0 * 1.5 / 3.0)
+
+    def test__set_consumer_to_demand(self):
+        """Test to set the consumer to the demand."""
+        # Act
+        consumers = self.controller._set_consumer_to_demand(datetime.now())
+        # Assert
+        self.assertEqual(consumers[self.consumer1.id][PROPERTY_HEAT_DEMAND], 1.0)
+        self.assertEqual(consumers[self.consumer1.id][PROPERTY_TEMPERATURE_RETURN], 20.0)
+        self.assertEqual(consumers[self.consumer1.id][PROPERTY_TEMPERATURE_SUPPLY], 30.0)
+
+        self.assertEqual(consumers[self.consumer2.id][PROPERTY_HEAT_DEMAND], 2.0)
+        self.assertEqual(consumers[self.consumer2.id][PROPERTY_TEMPERATURE_RETURN], 40.0)
+        self.assertEqual(consumers[self.consumer2.id][PROPERTY_TEMPERATURE_SUPPLY], 50.0)
+
+    def test__set_producers_based_on_priority(self):
+        """Test to set the producers based on priority."""
+        self.controller.consumers[0].get_heat_demand.return_value = 0.5
+        # Act
+        producers = self.controller._set_producers_based_on_priority(datetime.now())
+        # Assert
+        self.assertEqual(producers[self.producer1.id][PROPERTY_HEAT_DEMAND], 1.0)
+        self.assertEqual(producers[self.producer2.id][PROPERTY_HEAT_DEMAND], 1.5)
+
+    def test_run_time_step(self):
+        """Test to run the time step."""
+        # Act
+        result = self.controller.run_time_step(datetime.now())
+
+        # Assert
+        self.assertIn(self.producer1.id, result)
+        self.assertIn(self.producer2.id, result)
+        self.assertIn(self.consumer1.id, result)
+        self.assertIn(self.consumer2.id, result)
```

Comment:
Why not check the results?   this will be very valuable when the controller is improved further in the future  or the behavior/priorities are set in ESDL  (the results should not change)

Reply:
These are also tested in the other tests. It is usefull to add a test with actual parsing esdl and see what comes back

Reply:
added test with laoding of esdl and then creating controller and checking result for time step